src/DecayTreeVectorIntegral.cxx

```diff
@@ -150,4 +165,30 @@ const RealIntegralElement integral(const DecayTreeVectorIntegral& dtvi)
     return I;
 }
 
+//-------------------------
+const RealIntegralElement integral(const DecayTreeVectorIntegral& dtvi, const DecayTreeVector& dtv)
+{
+    auto dts = dtvi.decayTrees();
+
+    // find indices
+    std::vector<unsigned> indices;
+    for (auto dt : dtv) {
+        auto it = std::find(dts.begin(), dts.end(), dt);
+        if (it == dts.end())
+            throw exceptions::Exception("Trying to calculate integral for a DecayTree which is not in the DecayTreeVectorIntegral.", "integral");
+        indices.push_back(std::distance(dts.begin(), it));
+    }
+
+    RealIntegralElement I(0.);
+    for (auto i : indices) {
```

Comment:
since `integral(i, j)` also returns valid for `integral(i, i)`, this can be compressed:
```cpp
for (size_it i = 0; i < indices.size(); ++i)
    for (size_t j = i; j < indices.size(); ++j)
        I += dtvi.integral(i, j);
```